src/tsc/tsc.ts

```diff
@@ -174,6 +165,24 @@ namespace ts {
         }
     }
 
+    export function executeCommandLine(args: string[]): void {
+        if (args.length > 0 && args[0].charCodeAt(0) === CharacterCodes.minus) {
+            const firstOption = args[0].slice(args[0].charCodeAt(1) === CharacterCodes.minus ? 2 : 1).toLowerCase();
+            if (firstOption === "build" || firstOption === "b") {
+                return performBuild(args.slice(1));

```

Comment:
Can different composite projects have different values for the option?

Reply:
Yeah, in the same way that multiple projects can in theory have differing `extendedDiagnostics` settings, I guess.

Reply:
So what happens?  Separate trace for each project?

Reply:
I've just pushed a change so it should just make one trace for the whole build-mode invocation, since we can only really have one live cpu profile going at a time anyway. The `enableCPUProfiler` and `disableCPUProfiler` functions now return a `boolean` indicating if they actually (began to) enable/disable on that call.